src/http/routes.cc

```diff
@@ -98,10 +101,6 @@ future<std::unique_ptr<http::reply> > routes::handle(const sstring& path, std::u
             }
             auto r =  handler->handle(path, std::move(req), std::move(rep));
             return r.handle_exception(_general_handler);
-        } catch (const redirect_exception& _e) {
-            rep.reset(new http::reply());
-            rep->add_header("Location", _e.url).set_status(_e.status()).done(
-                    "json");
```

Comment:
Note to reviewers: by removing this redirect exception handling we delegate handling to `exception_reply` which handles `redirect_exception` in exactly the same way. However, in `exception_reply` registered exception handlers (via `routes::register_exception_handler`) take precedence. So in principle this could break some users? It seems low risk, but I'm happy to add it back if there is concern.

Reply:
I think this is fine and even better (if someone set an exception handler for redirect_exception, they wanted it to be used, not ignored). I don't see how any user will want to rely on a situation where if he set this handler explicitly, it will be ignored. But please correct me if I'm wrong and it's somehow more serious.

@amnonh maybe you can review too.